rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from json import JSONDecodeError
+from typing import TYPE_CHECKING
+
+import requests
+
+from rotkehlchen.assets.asset import UnderlyingToken
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import (
+    YEARN_VAULTS_V1_PROTOCOL,
+    YEARN_VAULTS_V2_PROTOCOL,
+    ChainID,
+    EvmTokenKind,
+    GeneralCacheType,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+YEARN_OLD_API = 'https://api.yearn.finance/v1/chains/1/vaults/all'
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_yearn_vaults(db: 'DBHandler') -> None:
+    """Query yearn API and ensure that all the tokens exist locally. If they exist but the protocol
+    is not the correct one, then the asset will be edited.
+
+    May raise:
+    - RemoteError
+    """
+    try:
+        response = requests.get(YEARN_OLD_API)
+        data = response.json()
+    except requests.exceptions.RequestException as e:
+        raise RemoteError(f'Failed to obtain yearn vault information. {str(e)}') from e
+    except (DeserializationError, JSONDecodeError) as e:
+        raise RemoteError(f"Failed to deserialize data from yearn's old api. {str(e)}") from e
+
+    if not isinstance(data, list):
+        raise RemoteError(f'Unexpected format from yearn vaults reponse. Expected a list, got {data}')  # noqa: E501
+
+    num_of_vaults = GlobalDBHandler().get_general_cache_values(
+        key_parts=[GeneralCacheType.YEARN_VAULTS],
+    )
+    if len(num_of_vaults) == 1 and int(num_of_vaults[0]) == len(data):
+        logging.debug(
+            f'Previous query of yearn vaults returned {num_of_vaults} vaults and last API'
+            f'response had the same amount of vaults. Skipping...',
```

Comment:
Skipping what? Writting to the DB? No you do write. What are you skipping. This log entry along with the condition above make no sense for anybody but you